pkg/reconciler/configuration/configuration.go

```diff
@@ -310,13 +311,19 @@ func (c *Reconciler) gcRevisions(ctx context.Context, config *v1alpha1.Configura
 		return nil
 	}
 
+	route, err := c.routeLister.Routes(config.Namespace).Get(config.Name)
```

Comment:
This is based on the assumption that the route name equals the configuration name, right? I think that might only coincidently be true for KnServices but is certainly not something we can depend on generally.

Reply:
Yes, you are correct. I assumed that `config` and `route` are always same. But now, updated to get the route by label value (`serving.knative.dev/route`).